semgrep-core/src/analyzing/CFG_build.ml

```diff
@@ -127,8 +129,9 @@ type cfg_stmt_result =
    *)
   | CfgFirstLast of F.nodei * F.nodei option
 
-let rec cfg_stmt : state -> F.nodei option -> stmt -> cfg_stmt_result =
- fun state previ stmt ->
+let rec cfg_stmt : Lang.t -> state -> F.nodei option -> stmt -> cfg_stmt_result
```

Comment:
Why not add Lang.t in state (which we kinda use like an environment)?
So we don't need to thread it all around (we already thread state all around).